### ISSUE TITLE
DataViews: Layout icon behaviour

### ISSUE THREAD

**Issue by JanaMW27** [WordPress/gutenberg#67418]
Currently, users can change the DataView layout through an icon that changes every time the user changes the DataView Layout.

We use:

**Table view: blockTable icon**
![Image](https://github.com/user-attachments/assets/aed9dc18-3df1-4a0c-827a-45c722fc2b12)

**Grid view: category icon**
![Image](https://github.com/user-attachments/assets/3d4d0501-4b0a-4f56-861a-c9bfe2a46ffc)

**List view: formatListBullets**
![Image](https://github.com/user-attachments/assets/602d7b92-1672-49ed-8b76-dd08722cc403)

**PROBLEM:**
If the user doesn't notice the icon change, for example, going from the Table View to the ListView, they may struggle to find the layout entry point to change the view again (since it is a different icon).

**SOLUTION:**
I would propose two options to solve this issue:

OPTION 1: 
To use a consistent icon as an entry point representing the layout concept. (this is a tricky icon, and I think, potentially, the blockable icon could be the one to use)

OPTION 2:
Introduce the icons in the layout dropdown to try to educate the user that they are related to the layout type and maybe make it easier for them to find the layout entry point.
Currently, the drop-down doesn't include the icons:
![Image](https://github.com/user-attachments/assets/e9979359-bf50-4e13-8986-bcd3a47cfcf1)

**Comment by matt-west** [comment]
> Introduce the icons in the layout dropdown to try to educate the user that they are related to the layout type and maybe make it easier for them to find the layout entry point.

+1. I think this would be a useful change.

We might need to find another way to signal the selected option in the dropdown though. Having both the active dot and the menu item icon could be a bit much.

**Comment by Mayank-Tripathi32** [comment]
@JanaMW27 @matt-west

I tried implementing option 2. Here’s how it looks:

![Image](https://github.com/user-attachments/assets/bfbcae34-96f9-4865-aa4e-f4633e229afb)

Additionally, I noticed that the pointer did not change to a cursor on hover. Shouldn’t it behave like this?

https://github.com/user-attachments/assets/1875ba2f-2777-4968-b495-f84e7c025afa

will open a sample PR to check out the changes aswell.

**Comment by jameskoster** [comment]
@Mayank-Tripathi32 thanks for looking into this and making a quick prototype, it helps get a better idea of how the proposed solution might work in practise.

I'm not totally convinced this is the correct approach though. The `Menu` component(s) intentionally do not provide affordances for indicating selection of regular menu items, this is partly to encourage the use of more semantically appropriate tools ([checkboxes](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-menu--with-checkboxes)/[radios](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-menu--with-radios)). I presume you've added a style override to change the icon color of the selected item? Ideally we avoid such overrides as they introduce inconsistencies and create backwards compatibility headaches down the road. Additionally color alone is not an adequate indication from an a11y perspective.

More generally, I believe this UI used a non-dynamic icon in the past. I acknowledge that's a bit duplicative given the view itself and the menu both communicate which layout is selected. But as mentioned in the OP a single icon that represents the layout concept is a bit tricky to pin down. I think that's one reason we find the dynamic icon to be a common design pattern (see: Dropbox, MacOS Finder).

**Comment by Mayank-Tripathi32** [comment]
@jameskoster Understood. In that case, we could consider suffixing the icon instead of using it as the primary indicator of selection. This approach would ensure semantic correctness while avoiding any backward compatibility concerns.

This solution leverages the current implementation of the radiobox and simply adds the icon to the side.

![Image](https://github.com/user-attachments/assets/6f1fc5d2-5c8c-48b7-b19d-35412951dea8)

**Comment by jameskoster** [comment]
@Mayank-Tripathi32 it can be explored, but it's a detail to look into separately considering to the widespread use of menus. 

One drawback to your suggestion is that decorative icons would occupy the same space as the chevrons that are used to indicate submenus. I worry this could lead to confusion about the nature of these icons. For example a consumer might use a chevron icon decoratively, which might feel broken in practice due to the 'missing' submenu.

**Comment by Mayank-Tripathi32** [comment]
True, that’s a valid point. Using the chevron icon decoratively in this manner could lead to user confusion. 

However, considering the current context, this might still be a feasible solution, as the need for decorative icons (next to labels) elsewhere seems unlikely.

**Comment by jameskoster** [comment]
It's a tricky problem to solve for sure. This exact use case (decorative icon + selected state) was discussed during implementation, but we couldn't find a satisfactory solution that didn't interfere with other features as noted above.

One option could be to display the selection indicator (radio/checkmark) on the right when a decorative icon is employed, but there would need to be conditions that prohibit mixtures. For example this would not be acceptable:

![Image](https://github.com/user-attachments/assets/195beb85-c866-4ff3-8eb4-c26de619ab94)

Whereas something like this could be:

![Image](https://github.com/user-attachments/assets/fbd16ec1-50ff-4b54-ab6c-0a85435efce5)

I don't know if it's possible to codify such logic though? :) 

By-and-by this use case feels like something to explore separately in a dedicated issue.

**Comment by ntsekouras** [comment]
Is there something actionable here considering that there was a unified icon in the past and updated to current state as better UX? --cc @WordPress/gutenberg-design

**Comment by jameskoster** [comment]
I'm not convinced there's anything urgent here.

**Comment by ntsekouras** [comment]
Thanks for the discussions all! I'll close for now and let's see if there will be future requests for this to reconsider.
